client.go

```diff
@@ -67,6 +67,14 @@ func (cl *Client) Do(c context.Context, req *Request) (*Response, error) {
 		if resp.response.StatusCode < 500 {
 			break
 		}
+
+		// close the response body before we lose our reference to it
+		if err = resp.response.Body.Close(); err != nil {
+			return nil, err
+		}
+
+		// wait before retrying
+		time.Sleep(req.backoffStrategy.waitDuration(i))
```

Comment:
What if a context deadline hits in the middle of this time.Sleep? Will that be a problem? Should we use a select statement with time.After instead?

Reply:
Good point. I was avoiding `time.After` on purpose so I wasn't unnecessarily creating a goroutine, but that's the reason to do it.

Reply:
Sweet.